### ISSUE TITLE
Keeping projects straight between multiple computers

### ISSUE THREAD

**Issue by malrice** [susanBuck/dwa15-fall2018#72]
Hello! I'm probably in a unique position where I work on my classwork on multiple computers (let's say weekday and weekend computers). Now, for other projects it has been fairly simple keeping projects straight, simply uploading to git and then syncing on the computers as necessary. For the laravel lessons and project 3, that's changed a bit. 

Basically, I started foobooks and p3 on my weekday computer, but now I'm trying to get it established on my weekend computer. One problem is that the git version doesn't sync the vendor and .env folders, and probably some other stuff that I'm missing. 

Any insights on can I do to get my weekend computer caught up?

**Comment by jessupjs** [comment]
Hi @malrice!

Have you been able to put Composer on your weekend computer? You may be able to imitate the deployment process by referring to [these notes](https://github.com/susanBuck/dwa15-fall2018/blob/master/laravel/deploy-to-digital-ocean.md).

**Comment by malrice** [comment]
Hi @jessupjs! 
Yes, I have installed composer and had been looking at those notes, but the I hit a problem when it comes to building out my project for development/my local server, verses the Digital Ocean server. So, the $ composer install --no-dev function wouldn't work because I AM using it for development, etc.

**Comment by susanBuck** [comment]
Working on multiple machines should be a relatively straightforward process, and it replicates the kind of things developers do on a team when you have multiple developers working on the same project.

Process:
1. Clone the project to the new machine
2. Create a new `.env` file by copying the `.env.example` and making any necessary adjustments
3. Run `composer update` 

You could run `composer install` with `--no-dev` omitted as @mdeforest noted - However because it's another local/development machine, it's better to run `composer update` which reads from `composer.json` instead of `composer.lock`.

To summarize:
+ `composer install --no-dev` : Install packages as listed in `composer.lock` and *exclude* dev dependencies. Typically run on production servers.
+ `composer install` : Install packages as listed in `composer.lock` and *include* dev dependencies. 
+ `composer update` : Install packages as listed in `composer.json`. Typically run on local/development servers.

Next week (7) - we're going to look closer at environments and packages and I'll talk in more detail about these commands and Composer.

**Comment by malrice** [comment]
Great! Glad to know I'm not missing anything. set up the vendor and the .env file, and everything seems to work!
